github/index.ts

```diff
@@ -121,35 +121,95 @@ export class GitHubClient {
     })
   }
 
+  private issueParts(issueUrl) {
+    const issURL = new url.URL(issueUrl)
+
+    // apiUrl: https://api.github.com/repos/bryanmacfarlane/quotes-feed/issues/9
+    // htmlUrl: https://github.com/bryanmacfarlane/quotes-feed/issues/9
+    const apiUrl = issURL.host.startsWith('api.')
+
+    const parts = issURL.pathname.split('/').filter(e => e)
+    const owner = parts[apiUrl ? 1 : 0]
+    const repo = parts[apiUrl ? 2 : 1]
+    const issue_number = parts[apiUrl ? 4 : 3]
+
+    return {
+      owner,
+      repo,
+      issue_number
+    }
+  }
+
+  public async ensureLabel(owner: string, repo: string, name: string, color: string): Promise<void> {
+    console.log(`ensure label ${owner} ${repo} ${name} ${color}`)
+
+    try {
+      const res = await this.octokit.issues.getLabel({
+        owner,
+        repo,
+        name
+      })
+
+      console.log(`${res.data} exists`)
+      return
+    } catch {
+      console.log(`Creating label ${name} ${color}`)
```

Comment:
throws on not existing.  not a fan of throwing if it successfully determined it doesn't exist